### PR TITLE
Hunter details

### PR DESCRIPTION
**Pull request by Davidde94**

### CODE REVIEW

File: Sources/SpartaLib/Facades/HuntersFacade.swift [hunk 1]
```diff
@@ -45,5 +45,49 @@ final class HuntersFacade: Facade {
...
+        self.executeRequest(request, response: response) { () -> SimpleResponse in
+
+            let identifier = request.parameters["identifier"]!
+            guard let hunter = try Hunter(identifier) else {
+                let response = SimpleResponse(
+                    message: "No hunter with that identifier",
+                    error: true,
+                    restResponseCode: .notFound
+                )
+                return response
```
Comment (RyanMKrol) [inline Sources/SpartaLib/Facades/HuntersFacade.swift]: As a rule I think we should try and return in one place, it'll make the code path easier to follow I feel.

File: Sources/SpartaLib/Facades/HuntersFacade.swift [hunk 2]
```diff
@@ -45,5 +45,49 @@ final class HuntersFacade: Facade {
...
+            }
+
+            let response = GenericResponse<Hunter>(
```
Comment (RyanMKrol) [inline Sources/SpartaLib/Facades/HuntersFacade.swift]: Like we could make one response, and set it accordingly, then just return last

File: Sources/SpartaLib/Facades/HuntersFacade.swift [hunk 3]
```diff
@@ -45,5 +45,49 @@ final class HuntersFacade: Facade {
...
+            return response
+        }
+    }
+    
+    func handleDeleteRequest(_ request: RouterRequest, response: RouterResponse) {
+        self.executeRequest(request, response: response) { () -> SimpleResponse in
+            
+            let identifier = request.parameters["identifier"]!
+            
+            guard try Hunter.existsForIdentifier(identifier) else {
```
Comment (RyanMKrol) [inline Sources/SpartaLib/Facades/HuntersFacade.swift]: Why are we doing exists for identifier here and not just making an object then deleting it?

File: Sources/SpartaLib/Routes.swift [hunk 4]
```diff
@@ -10,7 +10,11 @@ import Foundation
...
 class Routes {
     
-    static let CreateUserThirdParty = "/user/signup/thirdparty"
-    static let CreateUserEmail = "/user/signup/email"
+    static let CreateUserThirdParty = "/hunter/signup/thirdparty"
+    static let CreateUserEmail = "/hunter/signup/email"
+    
+    static let DeleteHunter = "/hunter/delete/:identifier"
+
+    static let GetHunterDetails = "/hunter/:identifier"
```
Comment (RyanMKrol) [inline Sources/SpartaLib/Routes.swift]: Just GetHunter would be my preference. What are you getting if not their details?

File: Sources/SpartaLib/Users/User.swift [hunk 5]
```diff
@@ -24,4 +24,8 @@ extension User {
         return "\(forename) \(surname)"
     }
     
+    static func existsForIdentifier(_ identifier: String) throws -> Bool {
```
Comment (RyanMKrol) [inline Sources/SpartaLib/Users/User.swift]: Why?

File: Sources/SpartaLib/Facades/HuntersFacade.swift (hunk 1 above)
Comment (Davidde94) [reply Sources/SpartaLib/Facades/HuntersFacade.swift]: Part of the Swift language pattern, `guards` *force* you to return.

File: Sources/SpartaLib/Users/User.swift (hunk 5 above)
Comment (Davidde94) [reply Sources/SpartaLib/Users/User.swift]: This is more efficient - if the hunter exists then we end up loading *all* of their details, and it more pressure on the DBMS to load a full row. This way we *may* end up with 2 requests in the case that a hunter does exist, but the first will be a simple count-based lookup that's highly optimised across most DBMSs (i.e. it doesn't need to load a full row, just know it exists).

File: Sources/SpartaLib/Facades/HuntersFacade.swift (hunk 3 above)
Comment (Davidde94) [reply Sources/SpartaLib/Facades/HuntersFacade.swift]: If it doesn't exist we send a 404

File: Sources/SpartaLib/Facades/HuntersFacade.swift (hunk 1 above)
Comment (RyanMKrol) [reply Sources/SpartaLib/Facades/HuntersFacade.swift]: I just think the code style is neater to return in one place, it reduces the likelihood to have a bunch of different routes through the code. You'd always end up in the same place with one return. Your call.

File: Sources/SpartaLib/Facades/HuntersFacade.swift (hunk 3 above)
Comment (RyanMKrol) [reply Sources/SpartaLib/Facades/HuntersFacade.swift]: Wouldn't the initialiser do the exact same thing? how is this: guard let hunter = try Hunter(identifier), different from Hunter.existsForIdentifier(identifier)

File: Sources/SpartaLib/Users/User.swift (hunk 5 above)
Comment (RyanMKrol) [reply Sources/SpartaLib/Users/User.swift]: I don't know that I agree with this. If you have to find the row based on ID, presumably you have a handle to all of the associated data to that handle.

**Pull request merged**